lib/src/network/load_balancing.rs

```diff
@@ -0,0 +1,59 @@
+use rand::random;
+
+use network::Backend;
+use sozu_command::config::LoadBalancingAlgorithms;
+
+use std::{ rc::Rc, cell::RefCell };
+use std::fmt::Debug;
+
+pub trait LoadBalancingAlgorithm: Debug {
+  fn next_available_backend(&mut self, backends: &Vec<Rc<RefCell<Backend>>>) -> Option<Rc<RefCell<Backend>>>;
+}
+
+#[derive(Debug)]
+pub struct RoundRobinAlgorithm {
+  pub next_backend: u32,
+}
+
+impl LoadBalancingAlgorithm for RoundRobinAlgorithm {
+
+  fn next_available_backend(&mut self , backends: &Vec<Rc<RefCell<Backend>>>) -> Option<Rc<RefCell<Backend>>> {
+    let sz = backends.len() as u32;
+    let res = backends.get(self.next_backend as usize)
+                      .map(|backend| (*backend).clone());
+
+    self.next_backend = if self.next_backend + 1 == sz {
+      0
+    } else {
+      self.next_backend + 1
+    };
+
+    res
+  }
+
+}
+
+impl RoundRobinAlgorithm {
+
+  fn new() -> Self {
+    Self {
+      next_backend: 0,
+    }
+  }
+
+}
+
+#[derive(Debug)]
+pub struct RandomAlgorithm;
+
+impl LoadBalancingAlgorithm for RandomAlgorithm {
+
+  fn next_available_backend(&mut self, backends: &Vec<Rc<RefCell<Backend>>>) -> Option<Rc<RefCell<Backend>>> {
+    let rnd = random::<usize>();
+    let idx = rnd % backends.len();
+
+    backends.get(idx)
```

Comment:
`Rng::choose()` might be a good choice there too https://docs.rs/rand/0.5.0/rand/trait.Rng.html#method.choose